pkg/operator/rules.go

```diff
@@ -63,10 +64,25 @@ func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, labelSelector
 	}, nil
 }
 
-func generateRulesConfiguration(ruleformat RuleConfigurationFormat, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
-	if ruleformat == PrometheusFormat {
-		// Unset partialResponseStrategy field.
-		for i := range promRule.Groups {
+func generateRulesConfiguration(ruleformat RuleConfigurationFormat, version semver.Version, namespace string, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
```

Comment:
The `version` parameter could be provided in `NewPrometheusRuleSelector()` instead. Also we could pass the version as a string and let `NewPrometheusRuleSelector()` build the semver.Version (since it already knows the flavor, e.g.  Thanos or Prometheus, it should good pick up the good default value).

Reply:
I initially added like that but I was thinking if version needs to be part of `PrometheusRuleSelector` struct

Reply:
Do we want version field to be part of  `PrometheusRuleSelecto struct` ?

Reply:
Discussed offline and decided to have version field in `PrometheusRuleSelector` struct